src/Utility/AnaSol.h

```diff
@@ -78,21 +80,25 @@ class AnaSol
   //! \brief The destructor frees the analytical solution fields.
   virtual ~AnaSol()
   {
-    if (scalSol)    delete scalSol;
-    if (scalSecSol) delete scalSecSol;
-    if (vecSol)     delete vecSol;
-    if (vecSecSol)  delete vecSecSol;
-    if (stressSol)  delete stressSol;
+    for (auto& it : scalSol)
+      delete it;
+    scalSol.clear();
+    for (auto& it : scalSecSol)
+      delete it;
+    scalSecSol.clear();
```

Comment:
The vector::clear calls (lines 85 and 88) are not needed since they will go out of scope anyway when leaving the destructor..